### PR TITLE
GT-2455 provide a Scarlet.Builder extension method to force the default platform

### PR DESCRIPTION
**Pull request by frett**

### CODE REVIEW

File: gto-support-scarlet/src/main/java/com/tinder/scarlet/internal/utils/RuntimePlatformInternals.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.tinder.scarlet.internal.utils;
+
+import androidx.annotation.NonNull;
+
+class RuntimePlatformInternals {
+    /** @noinspection KotlinInternalInJava*/
+    @NonNull
+    @SuppressWarnings("MethodName")
+    static RuntimePlatform Default() {
+        return new RuntimePlatform.Default();
```
Comment (frett) [inline gto-support-scarlet/src/main/java/com/tinder/scarlet/internal/utils/RuntimePlatformInternals.java:10]: `RuntimePlatform` is a Kotlin internal class in another library, we can still access it from Java to create the instance we need.

File: gto-support-scarlet/src/main/java/com/tinder/scarlet/internal/utils/RuntimePlatformInternals.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.tinder.scarlet.internal.utils;
+
+import androidx.annotation.NonNull;
+
+class RuntimePlatformInternals {
```
Comment (frett) [inline gto-support-scarlet/src/main/java/com/tinder/scarlet/internal/utils/RuntimePlatformInternals.java:5]: This is a package private Java class to hide it from the public API for library consumers

File: gto-support-scarlet/src/main/kotlin/com/tinder/scarlet/internal/utils/RuntimePlatformInternals.kt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package com.tinder.scarlet.internal.utils
+
+internal fun defaultPlatform() = RuntimePlatformInternals.Default()
```
Comment (frett) [inline gto-support-scarlet/src/main/kotlin/com/tinder/scarlet/internal/utils/RuntimePlatformInternals.kt:3]: we wrap the package-private java `RuntimePlatformInternals.Default()` in an internal Kotlin function, this allows us to access it from other packages in this module without exposing it on the public API

File: gto-support-scarlet/src/main/kotlin/org/ccci/gto/android/common/scarlet/ScarletBuilder.kt [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+ * Scarlet improperly detects desugared Android apps as the Java 8 runtime.
+ * This method will force the default platform for Scarlet to be used.
+ *
+ * See:
+ * https://github.com/Tinder/Scarlet/issues/235
+ * https://issuetracker.google.com/issues/342419066
+ */
+fun Scarlet.Builder.forceDefaultPlatform() = apply { platform = defaultPlatform() }
```
Comment (frett) [inline gto-support-scarlet/src/main/kotlin/org/ccci/gto/android/common/scarlet/ScarletBuilder.kt:15]: This is the only new thing exposed on the public API of this module

**Review by gyasistory — APPROVED** [review]

**Pull request merged**